File: flutter_app/lib/models/checkin.dart
import 'package:json_annotation/json_annotation.dart';

part 'checkin.g.dart';

@JsonSerializable()
class Checkin {
  final String id;
  @JsonKey(name: 'user_id')
  final String userId;
  @JsonKey(name: 'gym_id')
  final String gymId;
  @JsonKey(name: 'subscription_id')
  final String subscriptionId;
  @JsonKey(name: 'qr_token')
  final String qrToken;
  @JsonKey(name: 'checked_in_at')
  final String checkedInAt;
  @JsonKey(name: 'checked_out_at')
  final String? checkedOutAt;
  final String status;
  @JsonKey(name: 'daily_rate_paid')
  final double dailyRatePaid;
  @JsonKey(name: 'plan_tier')
  final String planTier;

  const Checkin({
    required this.id,
    required this.userId,
    required this.gymId,
    required this.subscriptionId,
    required this.qrToken,
    required this.checkedInAt,
    this.checkedOutAt,
    required this.status,
    required this.dailyRatePaid,
    required this.planTier,
  });

  factory Checkin.fromJson(Map<String, dynamic> json) =>
      _$CheckinFromJson(json);
  Map<String, dynamic> toJson() => _$CheckinToJson(this);
}

File: backend/app/schemas/checkin.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime
from decimal import Decimal
from uuid import UUID


class CheckinOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_id: UUID
    gym_id: UUID
    subscription_id: UUID
    qr_token: str
    checked_in_at: datetime
    checked_out_at: Optional[datetime] = None
    status: str
    daily_rate_paid: Decimal
    plan_tier: str


class CheckinRequest(BaseModel):
    gym_id: str
    qr_token: Optional[str] = None
